src/compiler/watchPublic.ts

```diff
@@ -434,15 +435,22 @@ namespace ts {
             const hasInvalidatedResolution = resolutionCache.createHasInvalidatedResolution(userProvidedResolution || changesAffectResolution);
             if (isProgramUptoDate(getCurrentProgram(), rootFileNames, compilerOptions, getSourceVersion, fileExists, hasInvalidatedResolution, hasChangedAutomaticTypeDirectiveNames, getParsedCommandLine, projectReferences)) {
                 if (hasChangedConfigFileParsingErrors) {
+                    if (reportFileChangeDetectedOnCreateProgram) {
+                        reportWatchDiagnostic(Diagnostics.File_change_detected_Starting_incremental_compilation);
+                    }
                     builderProgram = createProgram(/*rootNames*/ undefined, /*options*/ undefined, compilerHost, builderProgram, configFileParsingDiagnostics, projectReferences);
                     hasChangedConfigFileParsingErrors = false;
                 }
             }
             else {
+                if (reportFileChangeDetectedOnCreateProgram) {
+                    reportWatchDiagnostic(Diagnostics.File_change_detected_Starting_incremental_compilation);
+                }
                 createNewProgram(hasInvalidatedResolution);
             }
 
             changesAffectResolution = false; // reset for next sync
+            reportFileChangeDetectedOnCreateProgram = false;

```

Comment:
Can this be cleared in `updateProgramWithWatchStatus`?  I think that would make its lifetime/scope much clearer.

Reply:
I can clear it there instead, but I had been modelling `changesAffectResolution`.

My main concern was a case where it may be left over for another synchronize later, but I don't know why I thought that.

Reply:
Personally, I think it's clearer to set it and call it in the same place, but I'm also fine with following local conventions.  If this is where things are cleared after sync, then go for it.